vdp/pipeline/v1beta/pipeline.proto

```diff
@@ -1938,3 +1938,189 @@ message LookUpPipelineAdminResponse {
+  repeated ComponentRun component_runs = 1;
+
+  // The total number of component runs matching the request.
+  int64 total_count = 2;
+
+  // The current page number.
+  int32 page = 3;
+
+  // The number of items per page.
+  int32 page_size = 4;
+}
+
+// FileReference represents metadata for a file.
+message FileReference {
+  // Name of the file.
+  string name = 1;
+
+  // Format of the file (e.g., PDF, TXT, JPG).
+  string type = 2;
+
+  // Size of the file in bytes.
+  int64 size = 3;
+
+  // URL of the file (e.g., S3 URL).
+  string url = 4;
+}
+
+// PipelineRun represents a single execution of a pipeline.
+message PipelineRun {
+  // Unique identifier for the pipeline.
+  string pipeline_uid = 1;
+
+  // Unique identifier for each run.
+  string pipeline_trigger_uid = 2;
+
+  // Pipeline version used in the run.
+  string pipeline_version = 3;
+
+  // Current status of the run.
+  PipelineRunStatus status = 4;
+
+  // Origin of the run (e.g., Web click, API).
+  string source = 5;
+
+  // Time taken to complete the run in nanoseconds.
+  int64 total_duration = 6;
+
+  // Identity of the user who initiated the run.
+  string triggered_by = 7;
+
+  // Namespace of the pipeline (user or organization).
+  string namespace = 8;
+
+  // Input files for the run.
+  repeated FileReference inputs = 9;
+
+  // Output files from the run.
+  repeated FileReference outputs = 10;
+
+  // Snapshot of the pipeline recipe used for this run.
+  google.protobuf.Struct recipe_snapshot = 11;
+
+  // Time when the run was triggered.
+  google.protobuf.Timestamp triggered_time = 12;
+
+  // Time when the run started execution.
+  google.protobuf.Timestamp started_time = 13;
+
+  // Time when the run completed.
+  google.protobuf.Timestamp completed_time = 14;
+
+  // Error message if the run failed.
+  string error_msg = 15;
+
+  // Credits used of internal accounting metric.
+  double credits_used = 16;
+}
+
+// ComponentRun represents the execution details of a single component within a pipeline run.
+message ComponentRun {
+  // Links to the parent PipelineRun.
+  string pipeline_trigger_uid = 1;
+
+  // Unique identifier for each pipeline component.
+  string component_id = 2;
+
+  // Completion status of the component.
+  ComponentRunStatus status = 3;
+
+  // Time taken to execute the component in nanoseconds.
+  int64 total_duration = 4;
+
+  // Time when the component started execution.
+  google.protobuf.Timestamp started_time = 5;
+
+  // Time when the component finished execution.
+  google.protobuf.Timestamp completed_time = 6;
+
+  // Error message if the component failed.
+  string error_msg = 7;
+
+  // Input files for the component.
+  repeated FileReference inputs = 8;
+
+  // Output files from the component.
+  repeated FileReference outputs = 9;
+
+  // Credits used of internal accounting metric.
+  double credits_used = 10;
+}
+
+// PipelineRunStatus represents the possible states of a pipeline run.
+enum PipelineRunStatus {
```

Comment:
nit: I'd define this within the `PipelineRun` object or have a single `RunStatus` object that's shared between pipeline and component runs